Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -2,24 +2,66 @@ import BatteryInput from "~/components/batteryInput";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
+import Button from "~/components/Button/Button.tsx";
+import {useState} from "react";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
+  const [speed,setSpeed] = useState(null);
+  const [batteryPercent, changeBatteryPercent] = useState(0);
+  const [weather, weatherChange] = useState(0);
+  const[speedValid,setSpeedValid]= useState(false);
+  const[batteryValid,setBatteryValid]= useState(false);
```

Comment:
We're creating 2 state variables to check to see if the 2 inputs is valid. What if we had a form that had 50+ inputs, are we expecting to create 50+ corresponding valid states? There should be a better way to check if the inputs are valid without creating extra state.